core/k8s/provider.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package k8s
```

Comment:
Call this package deploymenttype or caas.

Reply:
We are moving away from caas towards k8s on our domains, why keep caas on `core/` then?

Reply:
This should be deploymenttype. It could extend past caas eventually.

Reply:
Done @manadart.

Reply:
This still shouldn't be in the package k8s or called k8s... This can now be generalised to be useful for any provider, it is just that we don't.